packages/web3-utils/src/converters.ts

```diff
@@ -362,6 +362,10 @@ export const toHex = (
 		return returnType ? 'bigint' : numberToHex(value);
 	}
 
+	if(value instanceof Uint8Array) {
```

Comment:
Could you please use `isUint8Array` from web3-validator because `instanceof` is not reliable at some rare edge cases. 

```suggestion
	if( isUint8Array(value) ) {
```

Thanks,

Reply:
isUint8Array implementation is incorrect as it returns false on buffer even though buffer is uint8array. 

In what cases does instanceof not work?

Reply:
`isUint8Array` checks for `data.constructor.name === 'Uint8Array'` in addition to checking the `instanceof`. So it is more inclusive than `instanceof`. 

The edge case were `instanceof` fails was reported at our library at: https://github.com/web3/web3.js/issues/6393. And it was also reported to be unreliable at one of our dependencies, where they applied our suggestion (https://github.com/paulmillr/noble-hashes/issues/25#issuecomment-1750106284) to fix it.

Reply:
🤦🏻‍♂️ 
![image](https://github.com/web3/web3.js/assets/8836210/2a084d3a-f990-4133-a58e-f9027c9afc3b)
```
--rant
Usually when people complain that `x instanceOf Uint8Array`, you are supposed to close the issue and tell them not to use a shitty engine that modifies globals (that also means not using jest+jsdom when you are developing a cross-platform library). 😅 

I'm really worried how everyone is ok with libraries and engines like jest/jsdome modifying globals and everybody else is applying hacks to make it work instead of urging people to not use projects like that
-- end rant
```
We cannot pretend Node.js doesn't exist and ignore Buffers so added commit which add support for Buffer in isUint8Array function and web3-validator.

Reply:
Thanks for the modifications. It looks good now that is not solely depending on `instanceof`.
Thanks also for using the same `data?.constructor?.name` for `Buffers`. 


A note and a question on the side: It would never be the intention to fix for jest/jsdom because nobody cares about it. And in the issue that was open at our library (#6393) the guy stated ["I have a Hyperledger Besu blockchain working on my VM with 4 nodes, built with Qorum Quickstart"](https://github.com/web3/web3.js/issues/6393). And the guy stated that he did not face the issue when he used ethers.js. Do you think that the issue that the guy reported was with jest/jsdom but falsely reported and we should had closed the issue stating "not to use a shitty engine that modifies globals"?

Reply:
Hard to say. My assumption is that Besu is using some shitty js engine to execute his javascript script so he should've open issue on besu repo saying that their js execution illegally modifies globals.

I would call bullshit that it works with ethers because: https://github.com/ethers-io/ethers.js/blob/ad5f1c5fc7b73cfb20d9012b669e9dcb9122d244/src.ts/utils/data.ts#L29 😂 😂

Reply:
Thanks @mpetrunic , And thanks for looking ethers code.

However, just for clarification, using `instanceof` is not reliable does not seem to be because of someone modifies globals. But because `Uint8Array` is "serializable, but not transferable" (https://github.com/web3/web3.js/issues/6393#issuecomment-1750095884).  

And following the provided links in the issues leads to [this article](https://medium.com/@simonwarta/limitations-of-the-instanceof-operator-f4bcdbe7a400) that mentions one of the cases for the failure of using `instanceof` when the guy is working on a browser extension. This is another example of the edge case that comes because of the nature of Uint8Array being "not transferable" as [mozilla documentation stated](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects).

Reply:
But doesn't that just mean that somebody illegally transferred "non-transferable" Uint8Array and then it was surprised that intanceof didn't work. Solution would then be to transfer underlying ArrayBuffer and on destination create new Uint8Array(arrayBuffer...) after which instanceof would work?

Reply:
But for jsdom, it's because they modified global Uint8Array: https://github.com/jsdom/jsdom/issues/1769

(for what it's worth happy-dom doesn't have those issues so if you are using vitest it's better to switch)